### PR TITLE
Issue 116  Adding persons (no UUIDs)

### LINKED ISSUE #116: [BUG] ArrayIndexOutOfBoundsException at AddPersonActivity.java:238
**Description of the bug**
ArrayIndexOutOfBoundsException at AddPersonActivity.java:238

### PR DESCRIPTION
**Pull request by mathisfchx**
### Description
Now we can add persons with the exact fields from the app(not tested with list for the moment)
Removed the unique_id (ICRC) check
Next PR => Having UUIDs generated when creating a person

### CODE REVIEW

File: Netw4ppl/app/build.gradle [hunk 1]
```diff
@@ -12,6 +12,7 @@ sonarqube {
 }
 
 android {
+    namespace 'netw4ppl.ines'
```
Comment (lduf) [inline Netw4ppl/app/build.gradle:15]: Can it be changed to SCAN ?

File: Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java [hunk 2]
```diff
@@ -75,26 +75,24 @@ protected void onCreate(Bundle savedInstanceState) {
...
-
                 Calendar calendar = Calendar.getInstance();
                 /* Ajout d'une nouvelle personne */
                 if (new_person) {
-
                     // ajout du champ date indispensable pour la database
                     SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
                     String date_creation = dateFormat.format(calendar.getTime());
                     person.putInfo("date", date_creation);
+                    System.out.println(person.getInfoByKey("date"));
```
Comment (lduf) [inline Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java]: TO be deleted before merging to preprod. Maybe you should add a tag to remember

File: Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java [hunk 3]
```diff
@@ -75,26 +75,24 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
                     // ajouter du champ d'application id
                     String application_id = MainActivity.mConfiguration.getApplicationId();
                     person.putInfo("application_id", application_id);
+                    System.out.println(person.getInfoByKey("application_id"));
```
Comment (lduf) [inline Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java]: TO be deleted before merging to preprod. Maybe you should add a tag to remember

File: Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java [hunk 4]
```diff
@@ -197,7 +199,7 @@ public static int getNextId(String key) {
      */
     public boolean verificationInputPerson() {
         boolean all_good = true;
-
+        System.out.println("verificationInputPerson");
```
Comment (lduf) [inline Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java]: TO be deleted before merging to preprod. Maybe you should add a tag to remember

File: Netw4ppl/app/src/main/java/netw4ppl/ines/ManagePersonsActivity.java [hunk 5]
```diff
@@ -31,14 +31,14 @@ public class ManagePersonsActivity extends AppCompatActivity {
     FloatingActionButton mButtonAdd;
     ListView mListView;
     SearchView mSearchBar;
-    public static ArrayList<Person> array_persons = new ArrayList<Person>();
+    public static ArrayList<Person> array_persons = new ArrayList<>();
```
Comment (lduf) [inline Netw4ppl/app/src/main/java/netw4ppl/ines/ManagePersonsActivity.java:34]: why not using  Person ArrayList ?

File: Netw4ppl/app/src/main/java/netw4ppl/ines/utils/AdapterViewFields.java [hunk 6]
```diff
@@ -293,10 +293,10 @@ public AdapterViewFields(Context context, ArrayList<Field> fields, boolean new_p
      * @param viewType an int representing the view type
      * @return a RecyclerView.ViewHolder object. The type of the object depends of the input type specified.
      */
-    @NonNull
     @Override
     public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
         View view;
+        System.out.println("viewType = " + viewType);
```
Comment (lduf) [inline Netw4ppl/app/src/main/java/netw4ppl/ines/utils/AdapterViewFields.java]: TO be deleted before merging to preprod. Maybe you should add a tag to remember

File: Netw4ppl/app/build.gradle (hunk 1 above)
Comment (mathisfchx) [reply Netw4ppl/app/build.gradle:15]: Not at the moment = it wil need a BIG refactoring

File: Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java (hunk 2 above)
Comment (mathisfchx) [reply Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java]: 👍🏼

File: Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java (hunk 3 above)
Comment (mathisfchx) [reply Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java]: 👍🏼

File: Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java (hunk 4 above)
Comment (mathisfchx) [reply Netw4ppl/app/src/main/java/netw4ppl/ines/AddPersonActivity.java]: 👍🏼

File: Netw4ppl/app/src/main/java/netw4ppl/ines/ManagePersonsActivity.java (hunk 5 above)
Comment (mathisfchx) [reply Netw4ppl/app/src/main/java/netw4ppl/ines/ManagePersonsActivity.java:34]: expected

File: Netw4ppl/app/src/main/java/netw4ppl/ines/utils/AdapterViewFields.java (hunk 6 above)
Comment (mathisfchx) [reply Netw4ppl/app/src/main/java/netw4ppl/ines/utils/AdapterViewFields.java]: 👍🏼

**Review by lduf — DISMISSED** [review]
🌍

**Pull request merged**